src/main/kotlin/org/neo4j/graphql/SchemaBuilder.kt

```diff
@@ -0,0 +1,299 @@
+
+            val relationTypes = objectTypeDefinitions
+                .filter { it.getDirective(DirectiveConstants.RELATION) != null }
+                .map { it.getDirective(DirectiveConstants.RELATION).getArgument(DirectiveConstants.RELATION_NAME).value.toJavaValue().toString() to it }
+                .toMap()
+
+            val nodeDefinitions: List<TypeDefinition<*>> = interfaceTypeDefinitions + objectTypeDefinitions
+            nodeDefinitions.forEach { createNodeMutation(it.getNodeType()!!) }
+
+            objectTypeDefinitions.forEach { createRelationshipMutations(it, objectTypeDefinitions, relationTypes) }
+            relationTypes.values.forEach { createNodeMutation(it.getNodeType()!!) }
+
+            recreateSchema()
+        }
+
+        private fun recreateSchema() {
+            val newSchemaDef = schemaDefinition.transform {
+                it.operationTypeDefinition(OperationTypeDefinition("query", TypeName(queryDefinition.name)))
+                    .operationTypeDefinition(OperationTypeDefinition("mutation", TypeName(mutationDefinition.name))).build()
+            }
+
+            typeDefinitionRegistry.remove(schemaDefinition)
+            typeDefinitionRegistry.add(newSchemaDef)
+            schemaDefinition = newSchemaDef
+        }
+
+        private fun mergeNeo4jEnhancements() {
+            val directivesSdl = javaClass.getResource("/neo4j.graphql").readText()
+            typeDefinitionRegistry.merge(SchemaParser().parse(directivesSdl))
+            neo4jTypeDefinitions
+                .forEach {
+                    val type = typeDefinitionRegistry.getType(it.typeDefinition)
+                        .orElseThrow { IllegalStateException("type ${it.typeDefinition} not found") }
+                            as ObjectTypeDefinition
+                    addInputType(it.inputDefinition, it.inputDefinition, type.fieldDefinitions)
+                }
+        }
+
+        private fun wireDefinedOperations(queryDefinition: ObjectTypeDefinition, isQuery: Boolean) {
+            for (fieldDefinition in queryDefinition.fieldDefinitions) {
+                QueryHandler.build(fieldDefinition, isQuery, metaProvider)?.let {
+                    addDataFetcher(queryDefinition.name, fieldDefinition.name, it)
+                }
+            }
+        }
+
+        fun addDataFetcher(type: String, name: String, dataFetcher: DataFetcher<Cypher>) {
+            wiringBuilder.type(type) { runtimeWiring -> runtimeWiring.dataFetcher(name, dataFetcher) }
+        }
+
+        /**
+         * add the given operation to the corresponding ObjectTypeDefinition
+         */
+        private fun mergeOperation(objectTypeDefinition: ObjectTypeDefinition, dataFetcher: BaseDataFetcher?)
+                : ObjectTypeDefinition {
+            if (dataFetcher == null) {
+                return objectTypeDefinition
+            }
+            if (objectTypeDefinition.fieldDefinitions.any { fd -> dataFetcher.fieldDefinition.name == fd.name }) {
```

Comment:
what if only some have one but others don't?

Reply:
this method is called per field on a root definition (mutation / query). If a dataFetcher was already defined we skip this field.